clash-for-windows: Add version 0.13.1

ZvonimirSun opened the pull request.

Ash258 reviewed (changes requested):
- Invalid manifest
- It is not clash, it is `clash-for-windows`
- URL in description is completely nonsense
- I do not think persist will work

ZvonimirSun commented:
- I have fixed the manifest.
- I changed the name to `clash-for-windows` now
- The description is copied from this cfw's github repo's 'About' content, and I have changed it now.
- The persist is to support the new portable mode according to the doc of cfw, and it seems to work properly now locally.

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+{
+    "homepage": "https://github.com/Fndroid/clash_for_windows_pkg",
```
This homepage is completely useless.

There is no information, nothing.

![image](https://user-images.githubusercontent.com/13260377/98087937-1d880500-1e81-11eb-8274-7a275f419919.png)

ZvonimirSun replied:
It's the only release place of this app, maybe it can be called a homepage? And there is a doc page using gitbook, that's the url in the description before. [https://docs.cfw.lbyczf.com/](https://docs.cfw.lbyczf.com/). Maybe I should put this in the homepage?

Ash258 replied:
I would not add this into official bucket.

There is no real homepage. Other "homepages" are in Chinese. This is unacceptable, as majority of users cannot properly use this.

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+{
+    "homepage": "https://github.com/Fndroid/clash_for_windows_pkg",
+    "description": "A Windows/macOS GUI based on Clash",
```
- MacOS is irrelevant
- what is clash?

ZvonimirSun replied:
Clash is a rule-based tunnel in Go according to the clash introduction, and it doesnot have a gui. https://github.com/Dreamacro/clash
I'll change the description later.

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+    "pre_install": [
+        "Remove-Item -Recurse \"$dir\\Uninstall Clash for Windows.exe\"",
+        "Get-ChildItem -Path \"$dir\\`$PLUGINSDIR\" -Depth \"2\" -Include \"*.7z\" | Move-Item -Destination \"$dir\\`$PLUGINSDIR\"",
+        "Expand-7zipArchive -Path \"$dir\\`$PLUGINSDIR\\app-64.7z\" -DestinationPath \"$dir\"",
```
This really sounds like it will work on 32bit systems..

ZvonimirSun replied:
> This really sounds like it will work on 32bit systems..

I'll test this later.

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+        "url": "https://github.com/Fndroid/clash_for_windows_pkg/releases/download/$version/Clash.for.Windows.Setup.$version.exe#/clash.7z",
+        "hash": {
+            "url": "https://github.com/Fndroid/clash_for_windows_pkg/releases/download/$version/sha256sum",
+            "regex": "\nexe: $sha256"
```
As soon as Excavator run first update it will invalidate the manfiest.

ZvonimirSun replied:
> As soon as Excavator run first update it will invalidate the manfiest.

But I find it works well locally. It can be installed properly and it didn't report any error. It seems that scoop only test the hash of the downloaded file and ignore the changes made on that later.

Ash258 replied:
`\n` is new line. `ConvertTo-Json` will replace it with new line, which lead into `regex: [ "", "exe:..."]`

ZvonimirSun replied:
> `\n` is new line. `ConvertTo-Json` will replace it with new line, which lead into `regex: [ "", "exe:..."]`

But I found the auto check treated this as an error, so I change it to `\n`

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+    ],
+    "pre_install": [
+        "Remove-Item -Recurse \"$dir\\Uninstall Clash for Windows.exe\"",
+        "Get-ChildItem -Path \"$dir\\`$PLUGINSDIR\" -Depth \"2\" -Include \"*.7z\" | Move-Item -Destination \"$dir\\`$PLUGINSDIR\"",
```
Completely nonsense step when you can simply pass destination path to the expand-7zip

ZvonimirSun replied:
> Completely nonsense step when you can simply pass destination path to the expand-7zip

Acturally, I'm not very familiar with this, and I just copy this from another manifest. I'll learn that.

Ash258 commented on bucket/clash-for-windows.json:
```diff
@@ -0,0 +1,29 @@
+        "Expand-7zipArchive -Path \"$dir\\`$PLUGINSDIR\\app-64.7z\" -DestinationPath \"$dir\"",
+        "Remove-Item -Recurse \"$dir\\`$PLUGINSDIR\""
+    ],
+    "persist": "data",
```
> The persist is to support the new portable mode according to the doc of cfw,

What docs?? Where are these docs?

ZvonimirSun replied:
> > The persist is to support the new portable mode according to the doc of cfw,
> 
> What docs?? Where are these docs?

Emm, it's a doc written by the author of the app with gitbook. https://docs.cfw.lbyczf.com/

Ash258 replied:
https://github.com/lukesampson/scoop-extras/pull/5049#discussion_r517188994

ZvonimirSun replied:
> [#5049 (comment)](https://github.com/lukesampson/scoop-extras/pull/5049#discussion_r517188994) Just close the PR and move to jp buckets

OK
